### PR TITLE
Added Login Controller

### PR DESCRIPTION
**Pull request by DikshantK2004**
Issue: #2 

## Short Description of what this resolves.
- Added router at '/api/user' which has 2 paths login and register
- Implemented the login route using 2 middleware for checking if the user exists and if the password is correct.
- On success, return object looks like:- 

image

### CODE REVIEW

File: src/middlewares/auth.middleware.js [hunk 1]
```diff
@@ -24,7 +23,57 @@ const ensureUniqueUsername = async (req, res, next) => {
     }
 };
 
+
+const findUser = async(req, res, next) =>{
+    try
+    {
+        const user = await User.findOne({email: req.body.email}).exec();
```
Comment (automatefreely) [inline src/middlewares/auth.middleware.js:30]: Here.

File: src/middlewares/auth.middleware.js [hunk 2]
```diff
@@ -24,7 +23,57 @@ const ensureUniqueUsername = async (req, res, next) => {
...
+    catch(err)
+    {
+        return response_500(res, "Error in User Login", err);
+    }
+}
+
+const validatePassword = async(req,res, next)=>{
+    try
+    {
+        const user = await User.findOne({email: req.body.email}).select('+password').exec();
```
Comment (automatefreely) [inline src/middlewares/auth.middleware.js]: Here.

File: src/controllers/auth.controller.js [hunk 3]
```diff
@@ -20,4 +20,28 @@ exports.signup = async (req, res) => {
     } catch (err) {
         return response_500(res, "Error creating user", err)
     }
-};
\ No newline at end of file
+};
+
+exports.login = async(req, res) =>
+{
+    const user = await User.findOne({email: req.body.email}).exec();
```
Comment (automatefreely) [inline src/controllers/auth.controller.js]: You are querying the user base on email **3** time. In controller and twice in middleware.

**Review by automatefreely — COMMENTED** [review]
Try sending the queried user by storing it in req.

File: src/middlewares/auth.middleware.js [hunk 4]
```diff
@@ -24,7 +23,61 @@ const ensureUniqueUsername = async (req, res, next) => {
     }
 };
 
+
+const findUser = async(req, res, next) =>{
+    try
+    {
+        const user = await User.findOne({email: req.body.email}).exec();
```
Comment (automatefreely) [inline src/middlewares/auth.middleware.js:30]: Here you need to select password.

Comment (DikshantK2004) [reply src/middlewares/auth.middleware.js:30]: But then if it is a wrong password, I won't be able to detect it correctly as I will get null user for both wrong password and user not found. Shouldn't they be different?

Comment (automatefreely) [reply src/middlewares/auth.middleware.js:30]: you are right.

**Pull request merged**